pypesto/hierarchical/solver.py

```diff
@@ -262,25 +307,21 @@ def fun(x):
         pypesto_problem = Problem(
             objective, lb=lb, ub=ub, x_names=x_names, **self.problem_kwargs
         )
-
-        if self.x_guesses is not None:
-            pypesto_problem.set_x_guesses(
-                self.x_guesses[:, pypesto_problem.x_free_indices]
-            )
-        else:
-            pypesto_problem.set_x_guesses(
-                [list(problem.get_dummy_values(scaled=False).values())]
-            )
+        pypesto_problem.set_x_guesses(
+            x_guesses[:, pypesto_problem.x_free_indices]
+        )
 
         # perform the actual optimization
         result = minimize(pypesto_problem, **self.minimize_kwargs)
-
         best_par = result.optimize_result.list[0]['x']
 
-        if (np.isclose(best_par, lb) | np.isclose(best_par, ub)).any():
+        # Check if any optimized parameter is on the dummy bound
+        # If so, raise an error
+        if any(best_par == self.dummy_lb) or any(best_par == self.dummy_ub):
             raise RuntimeError(
-                "Active bounds in inner problem optimization. This can result "
-                "in incorrect gradient computation for the outer parameters."
+                "An optimal inner parameter is on the dummy bound of numerical optimization. "
+                "This means the optimal inner parameter is either "
+                f"extremely large ({self.dummy_ub}) or extremely small ({self.dummy_lb})."
```

Comment:
Makes sense, but `if any(best_par <= self.dummy_lb) or any(best_par >= self.dummy_ub)` will trigger somewhat often if a user chooses bounds larger than the dummy bounds, right? So the `==` case will fail too (very rarely...). Why not either prevent the user from choosing more-extreme-than-dummy bounds, or use the user's bounds in this check?

Reply:
Alright, then perhaps emit a warning if a user specifies the same bound(s) as  `dummy_(l/u)b`, because then it might be mechanistic, and might hit the bounds often as you say.